File: server/Application/Templates/TemplateResponse.cs
using System.ComponentModel;
using Application.Common;
using Microsoft.AspNetCore.Http;

namespace Application.Templates;

/// <summary>
///     The template DTO.
/// </summary>
[DisplayName("TemplateResponse")]
public class TemplateResponse
{
    public Guid Id { get; set; }
    public Guid? ProjectId { get; set; }
    public string? Name { get; set; }
    public string? Description { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.Now;
    public DateTime UpdatedAt { get; set; } = DateTime.Now;
    public ICollection<TemplateStepResponse> Steps { get; set; } = new List<TemplateStepResponse>();
}

/// <summary>
///     The template step DTO.
/// </summary>
[DisplayName("TemplateStepResponse")]
public class TemplateStepResponse
{
    public Guid Id { get; set; }
    public string? Name { get; set; }
    public int Duration { get; set; }
    public int Position { get; set; }
    public MediaResponse? Media { get; set; }
}

/// <summary>
///     The template step DTO.
/// </summary>
[DisplayName("TemplateStepDto")]
public class TemplateStepDto
{
    public Guid Id { get; set; }
    public string? Name { get; set; }
    public int Duration { get; set; }
    public int Position { get; set; }
    public IFormFile? Media { get; set; }
}

File: server/admin-board/src/lib/api/dto/TemplateResponse.ts
export interface Template {
  id: string;
  projectId?: string | null;
  name: string;
  description: string;
  createdAt: string;
  steps: TemplateStep[];
}

export interface TemplateStep {
  id: string;
  name: string;
  position: number;
  duration: number;
  media?: any;
}

export interface TemplateDto {
  name: string;
  description: string;
  projectId?: string | null;
}

export interface EditTemplateDto {
  name: string;
  description: string;
  steps: TemplateStep[];
}
